test/checkoutput/main.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// verifies the output of a command with the expected output
+// There are 2 states : start and Parsing.
+// start ---Run:--> Parsing (Capture Run)
+// Parsing ---Run:---> Parsing (Reinitialize Run)
+// Parsing ---Expected output:---> start (Capture verify and goto start)
+/*
+The parsing is done using an Finite State Machine(FSM). The FSM has 2 states Start and Parsing. The initial state is Start. The parsing logic is summarized in the following table.
+
+| State | Input | Action | Output State
+|------ |-------|--------|-------------|
+| Start | Run Block | Record the command| Parsing|
+| Parsing | Run Block | Record the command| Parsing|
+| Parsing | Verify Block | Record the output | Start|
+
+An alternative is to use a markdown parser but that would help us only in the Action column and it would introduce external dependencies.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+var (
+	vlevel = flag.Int("v", 1, "specify the log verbosity level")
+)
+
+// FSM State of the parsing.
+type state uint32
+
+const (
+	// Initial state waiting for Run: cmd
+	start state = iota
+	// Middle of parsing Run/Verify
+	parsing
+)
+
+const (
+	runCmd = iota
+	verifyCmd
+)
+
+// state-command combination
+type commandState struct {
+	state   state
+	command uint32
+}
+
+// Run-verify combo
+type runVerify struct {
+	run    string
+	verify string
+	dir    string
+}
+
+func (rv *runVerify) validate() error {
+
+	fcmd := strings.Fields(rv.run)
+	binary := fcmd[0]
+	args := fcmd[1:]
+	cmd := exec.Command(binary, args...)
+	cmd.Dir = rv.dir
+	// TODO: Replace with CombinedOutput() ?
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to execute command %v \n error: \n %v \ncommand-output: %s", rv.run, fmt.Sprint(err), output)
+	}
+
+	// TODO: Currently we capture the output of just the stdout. Logs extensively write to
+	// stderr, we need to come up with the right logic for validation
+	if string(output) == rv.verify {
+		fmt.Printf("Validation Success: command : %v \n Matched text: %v\n", rv.run, rv.verify)
```

Comment:
All of the fmt.Printf calls should be `klog.InfoS` calls, potentially with `V()` if less important.

Reply:
I had pushed a commit that did not have the change. The latest code has the klogs.

Reply:
Fixed.